Web3Swift/Blocks/BlockHash.swift

```diff
@@ -0,0 +1,26 @@
+//
+// This source file is part of the Web3Swift.io open source project
+// Copyright 2019 The Web3Swift Authors
+// Licensed under Apache License v2.0
+//
+// BlockHash.swift
+//
+// Created by Vadim Koleoshkin on 14/05/2019
+//
+
+import Foundation
+
+/** Standard ethereum block hash */
+public protocol BlockHash: BytesScalar {
+    
+    /**
+    Block representation of ethereum block hash
+
+    - parameters:
+     - network: `Network` to fetch from JSON-RPC node
+
+    - returns:
+    `Block` object
+    */
+    func block(network: Network) throws -> Block
```

Comment:
`Network` in relationship to the `BlockHash` seems to me like something that should be an implementation detail rather than a part of the interface. 
Are you sure you intend to query blocks by hash on a variety of networks?

Reply:
There are two assumptions around this issue: 
1) `BlockHash` and `TransactionHash` both could exist (and obtained) from different sources that do not depend on `Network`. That's why I moved it from constructor itself 
2) Getting `Receipt`, `Transaction`, `Block` require `Network` in the current implementation and they could be obtained from different `Networks`. 
Leaving both objects without the ability to fetch related entities will make it DTO and we will need to create separate object to fetch them. 
I am almost fine to make it work in this way for now and then move this another Domain Object

Reply:
Could you please draft up an example of an integration test where `BlockHash` or a `TransactionHash` would originate from a source not dependent on a `Network` and would require `Network` as a dynamic dependency.

Reply:
There are several examples of this: 
1) `TransactionHash` obtained by third-party API (application backend for instance) and not from the node itself. But once we need to check transaction status we will need `Network`
2) In this way libraries build on top of Web3Swift could use this protocol to originate `TransactionHash` without dependence on `Network`
3) There are many services that provide `Cloud wallets` for a user (https://fortmatic.com/, https://bitski.com/) and their API is different from JSON-PRC but they return `TransactionHash` that could later be used for fetching `Receipt`.

Reply:
@biboran but maybe you are right, `TransactionHash` itself will most likely be obtained from `Network`. But then the question is if it is right to always construct `Receipt`, `Block`, `Transaction` with `Network`?

Reply:
I think it is appropriate to construct such an object chain by passing the `Network` implicitly rather than requiring the client to specify the `Network` each time. You can create an additional implementation for `TransactionHash` which will allow you to set its properties as closures in case you want to specify how to it returns the `Transaction` explicitly (although I am not sure its a good design).

Reply:
So let's return make `Network` passed explicitly and then think of how we could make it configurable?

Reply:
Here is the design I am proposing:
```swift
public protocol BlockHash: BytesScalar { 
	func block() throws -> Block
}

public final class EthBlockHash: BlockHash {
	...

	init(id: BytesScalar, network: Network) {
	...
	}

	func block() throws -> Block {
		return EthBlock(
			hash: self,
			network: network
		)
	}
}
```